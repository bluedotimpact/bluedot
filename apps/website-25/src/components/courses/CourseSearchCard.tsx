import React from 'react';
import clsx from 'clsx';
import { Tag } from '@bluedot/ui';
import { H3, P } from '../Text';

export type CourseSearchCardProps = React.PropsWithChildren<{
  // Required
  title: string,
  url: string,
  // Optional
  className?: string,
  description?: string,
  duration?: string,
}>;

export const CourseSearchCard: React.FC<CourseSearchCardProps> = ({
  children,
  className,
  description,
  duration,
  title,
  url,
}) => {
  return (
    <a
      href={url}
      className={clsx(
        'course-search-card flex flex-row container-lined p-6 transition-transform duration-200 hover:scale-[1.01] hover:container-elevated',
        className,
      )}
    >
      <div className="course-search-card__content block md:flex gap-space-between w-full">
        <div className="course-search-card__text">
          <H3 className="course-search-card__title mb-6">
            {title}
          </H3>
          {description && (
            <P className="course-search-card__description text-sm overflow-hidden text-bluedot-black text-ellipsis mb-6">
              {description}
            </P>
          )}
          {duration && (
            <Tag>{duration}</Tag>
          )}
        </div>
      </div>
      {children}
    </a>
  );
};
