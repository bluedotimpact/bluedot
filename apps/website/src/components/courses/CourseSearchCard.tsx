import React from 'react';
import clsx from 'clsx';
import { ClickTarget, Tag } from '@bluedot/ui';
import { FaStar } from 'react-icons/fa6';
import { H3, P } from '../Text';

export type CourseSearchCardProps = React.PropsWithChildren<{
  // Required
  title: string,
  url: string,
  // Optional
  className?: string,
  description?: string,
  cadence?: string,
  courseLength?: string,
  averageRating?: number | null;
  imageSrc?: string,
}>;

const displayAverageRating = (rating?: number | null) => (rating ? rating.toFixed(1) : '');

export const CourseSearchCard: React.FC<CourseSearchCardProps> = ({
  children,
  className,
  description,
  cadence,
  courseLength,
  averageRating,
  imageSrc,
  title,
  url,
}) => {
  const defaultImageSrc = '/images/courses/default.jpg';

  return (
    <ClickTarget
      url={url}
      className={clsx(
        'course-search-card flex flex-col container-lined p-6 max-w-[828px] size-full transition-transform duration-200 hover:scale-[1.01] hover:container-elevated',
        className,
      )}
    >
      <div className="course-search-card__image-container h-[200px] w-full mb-4">
        <img className="course-search-card__image size-full object-cover rounded-lg" src={imageSrc || defaultImageSrc} alt={title} />
      </div>
      <div className="course-search-card__content flex flex-col flex-1">
        <div className="course-search-card__text flex-1">
          <H3 className="course-search-card__title mb-3">
            {title}
          </H3>
          {description && (
            <P className="course-search-card__description text-bluedot-black">
              {description}
            </P>
          )}
        </div>
        {typeof averageRating === 'number' && (
          <div className="course-search-card__footer flex justify-end mt-4">
            <div className="course-card__footer-left text-size-xs text-bluedot-black gap-1 flex flex-col items-end">
              <span className="course-card__rating flex gap-[3px] items-start font-bold">
                {displayAverageRating(averageRating)}{' '}
                <FaStar className="mt-px" />
              </span>
            </div>
          </div>
        )}
      </div>
      {children}
    </ClickTarget>
  );
};
