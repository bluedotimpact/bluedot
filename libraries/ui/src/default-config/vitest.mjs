/* eslint-disable import/no-extraneous-dependencies */
import { defineConfig, defaultExclude } from 'vitest/config';
import react from '@vitejs/plugin-react';
import * as dotenv from 'dotenv';

/**
 * @param {import('vitest/config').UserConfig} [config] 
 * @returns {import('vitest/config').UserConfig}
 */
export const withDefaultBlueDotVitestConfig = async (config) => defineConfig({
  ...config,
  plugins: [react(), ...(config?.plugins ?? [])],
  test: {
    environment: 'happy-dom',
    env: dotenv.config({ path: '.env.test' }).parsed,

    ...config?.test,

    exclude: [...defaultExclude, '**/.{turbo,next}/**', ...(config?.test?.exclude ?? [])],
  },
});
