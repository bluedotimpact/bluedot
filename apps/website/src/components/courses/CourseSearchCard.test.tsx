import { describe, expect, test } from 'vitest';
import { render } from '@testing-library/react';
import { CourseSearchCard } from './CourseSearchCard';

describe('CourseSearchCard', () => {
  const defaultProps = {
    title: 'Title',
    url: 'https://bluedot.org/courses/what-the-fish',
  };

  test('renders default as expected', () => {
    const { container } = render(
      <CourseSearchCard {...defaultProps} />,
    );
    expect(container).toMatchSnapshot();
  });

  test('renders with optional args', () => {
    const { container } = render(
      <CourseSearchCard
        {...defaultProps}
        description="A short description of the course"
        courseLength="5 hours"
        cadence="Daily"
        averageRating={4.6534563}
        imageSrc="/courses/what-the-fish/image.png"
      />,
    );
    expect(container).toMatchSnapshot();
  });
});
