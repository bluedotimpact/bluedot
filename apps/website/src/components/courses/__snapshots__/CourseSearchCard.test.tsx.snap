// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`CourseSearchCard > renders default as expected 1`] = `
<div>
  <a
    class="course-search-card flex sm:flex-row flex-col gap-6 container-lined p-6 max-w-[828px] size-full transition-transform duration-200 hover:scale-[1.01] hover:container-elevated"
    href="https://bluedot.org/courses/what-the-fish"
    tabindex="0"
  >
    <div
      class="course-search-card__image-container h-[240px] sm:w-fit w-full"
    >
      <img
        alt="Title"
        class="course-search-card__image size-full object-cover rounded-lg"
        src="/images/courses/default.jpg"
      />
    </div>
    <div
      class="course-search-card__content flex flex-col justify-between gap-space-between flex-1"
    >
      <div
        class="course-search-card__text"
      >
        <h3
          class="bluedot-h3 not-prose course-search-card__title mb-4"
        >
          Title
        </h3>
      </div>
      <div
        class="course-search-card__footer flex flex-wrap gap-4"
      >
        <div
          class="flex gap-2"
        />
      </div>
    </div>
  </a>
</div>
`;

exports[`CourseSearchCard > renders with optional args 1`] = `
<div>
  <a
    class="course-search-card flex sm:flex-row flex-col gap-6 container-lined p-6 max-w-[828px] size-full transition-transform duration-200 hover:scale-[1.01] hover:container-elevated"
    href="https://bluedot.org/courses/what-the-fish"
    tabindex="0"
  >
    <div
      class="course-search-card__image-container h-[240px] sm:w-fit w-full"
    >
      <img
        alt="Title"
        class="course-search-card__image size-full object-cover rounded-lg"
        src="/courses/what-the-fish/image.png"
      />
    </div>
    <div
      class="course-search-card__content flex flex-col justify-between gap-space-between flex-1"
    >
      <div
        class="course-search-card__text"
      >
        <h3
          class="bluedot-h3 not-prose course-search-card__title mb-4"
        >
          Title
        </h3>
        <p
          class="bluedot-p not-prose course-search-card__description overflow-hidden text-bluedot-black text-ellipsis mb-6"
        >
          A short description of the course
        </p>
      </div>
      <div
        class="course-search-card__footer flex flex-wrap gap-4"
      >
        <div
          class="flex gap-2"
        >
          <span
            class="tag container-lined inline-flex items-center px-4 py-2 text-xs font-semibold text-color-secondary-text course-search-card__level"
            role="status"
          >
            Beginner
          </span>
          <span
            class="tag container-lined inline-flex items-center px-4 py-2 text-xs font-semibold text-color-secondary-text course-search-card__duration"
            role="status"
          >
            Weekly
          </span>
        </div>
        <div
          class="course-card__footer-left ml-auto text-size-xs text-bluedot-black gap-1 flex flex-col items-end justify-end"
        >
          <span
            class="course-card__rating flex gap-[3px] items-start font-bold pb-2"
          >
            4.7
             
            <svg
              class="mt-px"
              fill="currentColor"
              height="1em"
              stroke="currentColor"
              stroke-width="0"
              viewBox="0 0 576 512"
              width="1em"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M316.9 18C311.6 7 300.4 0 288.1 0s-23.4 7-28.8 18L195 150.3 51.4 171.5c-12 1.8-22 10.2-25.7 21.7s-.7 24.2 7.9 32.7L137.8 329 113.2 474.7c-2 12 3 24.2 12.9 31.3s23 8 33.8 2.3l128.3-68.5 128.3 68.5c10.8 5.7 23.9 4.9 33.8-2.3s14.9-19.3 12.9-31.3L438.5 329 542.7 225.9c8.6-8.5 11.7-21.2 7.9-32.7s-13.7-19.9-25.7-21.7L381.2 150.3 316.9 18z"
              />
            </svg>
          </span>
        </div>
      </div>
    </div>
  </a>
</div>
`;
